File: testbed/stages/javascript/compressed.js



class C1 {
    constructor(v1, v2,
                v3, v4,
                v5 = 0.0) {
        this.v1 = v1;
        this.v2 = v2;
        this.v3 = v3;
        this.v4 = v4;
        this.v5 = v5;
    }
}


class C2 {
    constructor(v6, v7) {
        this.v8  = '';
        this.v6  = v6;
        this.v7  = v7;
        this.v9  = [];
        this.v10 = 0.0;
        this.v11 = 0.0;
        this.v12 = 'PENDING';
    }
}


function f1(v13) {
    if (!v13.v6) return false;
    if (!v13.v7) return false;
    if (!v13.v9 || v13.v9.v14 === 0) return false;
    return true;
}


function f2(v13) {
    let v15 = 0.0;
    for (const v16 of v13.v9) {
        const v17 = v16.v4 * v16.v3;
        v15 += v17;
    }
    return v15;
}


function f3(v13, v18) {
    const v19 = v13.v10
                * (v18 / 100.0);
    v13.v11 =
        v13.v10 - v19;
}


function f4(v13, v20) {
    v13.v8  = v20;
    v13.v12 = 'CONFIRMED';
}


function f5(v13, v21,
            v22) {
    v13.v12 = 'CANCELLED';
    void v21;
    void v22;
}

v23.v24 = {
    C1,
    C2,
    f1,
    f2,
    f3,
    f4,
    f5,
};

File: testbed/stages/cpp/compressed.cpp




#include <string>
#include <vector>
#include <stdexcept>


struct C1 {
    v1::v2 v3;
    v1::v2 v4;
    int         v5;
    float       v6;
    float       v7;
};


struct C2 {
    v1::v2                v8;
    v1::v2                v9;
    v1::v2                v10;
    v1::v11<C1> v12;
    float                      v13;
    float                      v14;
    v1::v2                v15;
};



bool f1(C2 v16) {
    if (v16.v9.v17()) return false;
    if (v16.v10.v17()) return false;
    if (v16.v12.v17())          return false;
    return true;
}


float f2(C2 v16) {
    float v18 = 0.0f;
    for (auto& v19 : v16.v12) {
        float v20 = v19.v6
                               * v19.v5;
        v18 += v20;
    }
    return v18;
}



void f3(C2& v16,
                                     float v21) {
    float v22 = v16.v13
                                 * (v21 / 100.0f);
    v16.v14 =
        v16.v13 - v22;
}



void f4(C2& v16,
                                     v1::v2 v23) {
    v16.v8   = v23;
    v16.v15 = "CONFIRMED";
}



void f5(C2& v16,
                                   v1::v2 v24,
                                   v1::v2 v25) {
    v16.v15 = "CANCELLED";
    (void)v24;
    (void)v25;
}
